SDK-2825: Rework the documentation for evaluator (#1931)
spryker/spryker-docs: docs/scos/dev/guidelines/keeping-a-project-upgradable/upgradability-guidelines/multidimensional-array.md

template: howto-guide-template
---

> This check checks that project doesn't use the deeply nested multidimensional arrays in dependency providers for not overloading it with complicated logic.

Reviewer: ```suggestion
This check checks that project doesn't use the deeply nested multidimensional arrays in dependency providers, in order to not overload it with complicated logic.
```

## Problem description

> If a plugins stack is used on the project level, not all structures are necessarily needed. Deeply nested multidimensional arrays make configuration hard to upgrade.

Reviewer: ```suggestion
If a plugins stack is used on the project level, not all structures are necessarily required. Deeply nested multidimensional arrays make configuration hard to upgrade.
```

> This check verifies that multidimensional arrays have maximum of two levels of nesting inside.

Reviewer: ```suggestion
This check verifies that multidimensional arrays have a maximum of two levels of nesting inside.
```

> ## Example of evaluator error message

Reviewer: ```suggestion
## Example of an Evaluator error message
```
